src/types/arconnect/arconnect.types.ts

```diff
@@ -0,0 +1,268 @@
+// Copied and adapted from https://www.npmjs.com/package/arconnect?activeTab=code
+
+import type { SignatureOptions } from "arweave/node/lib/crypto/crypto-interface";
+import type Transaction from "arweave/node/lib/transaction";
+
+export abstract class ArConnect {
```

Comment:
@7i7o I copied this type from ArConnect in order to have a compile-time check that makes sure we always have compatible interfaces. There are 3 issues though, `connect`, `dispatch` and `signDataItem`. More details in `index.ts`...